src/universal_transfer_operator/data_providers/database/google/bigquery.py

```diff
@@ -155,6 +238,208 @@ def truncate_table(self, table):
+        native_support_kwargs.update(native_support_kwargs)
+
+        # Since bigquery has other options besides used here, we need to expose them to end user.
+        # https://cloud.google.com/bigquery/docs/reference/rest/v2/Job#JobConfigurationLoad
+        load_job_config.update(kwargs)
+
+        job_config = {
+            "jobType": "LOAD",
+            "load": load_job_config,
+            "labels": {"target_table": target_table.name},
+        }
+
+        self.hook.insert_job(
+            configuration=job_config,
+        )
+
+    def load_s3_file_to_table(
+        self,
+        source_file: File,
+        target_table: Table,
+        **kwargs,
+    ):
+        """
+        Load content of multiple files in S3 to output_table in Bigquery by using a datatransfer job
+        Note - To use this function we need
+        1. Enable API on Bigquery
+        2. Enable Data transfer service on Bigquery, which is a chargeable service
+        for more information refer - https://cloud.google.com/bigquery-transfer/docs/enable-transfer-service
+        :param source_file: Source file that is used as source of data
+        :param target_table: Table that will be created on the bigquery
+        :param if_exists: Overwrite table if exists. Default 'replace'
+        """
+        native_support_kwargs = self.transfer_params.s3_transfer_parameters or {}
+
+        project_id = self.get_project_id(target_table)
+        transfer = S3ToBigqueryDataTransfer(
+            target_table=target_table,
+            source_file=source_file,
+            project_id=project_id,
+            native_support_kwargs=native_support_kwargs,
+            **kwargs,
+        )
+        transfer.run()
+
+    def get_project_id(self, target_table) -> str:
+        """
+        Get project id from the hook.
+        :param target_table: table object that the hook is derived from.
+        """
+        try:
+            return str(self.hook.project_id)
+        except AttributeError as exe:
+            raise DatabaseCustomError(f"conn_id {target_table.conn_id} has no project id") from exe
+
+    def load_local_file_to_table(
+        self,
+        source_file: File,
+        target_table: Table,
+        if_exists: LoadExistStrategy = "replace",
+        **kwargs,
+    ):
+        """
+        Transfer data from local to bigquery
+
+        :param source_file: Source file that is used as source of data
+        :param target_table: Table that will be created on the bigquery
+        :param if_exists: Overwrite table if exists. Default 'replace'
+        """
+        # We need to maintain file_type to biqquery_format and not use NATIVE_PATHS_SUPPORTED_FILE_TYPES
+        # because the load_table_from_file expects 'JSON' value for ndjson file.
+        file_types_to_bigquery_format = {
+            FileType.CSV: "CSV",
+            FileType.NDJSON: "JSON",
+            FileType.PARQUET: "PARQUET",
+        }
```

Comment:
Can we move this to a constant because I see a couple of occurrences for this in the PR?

Reply:
`file_types_to_bigquery_format` here is class variable for generic `BigqueryDataProvider`